helper/variables/parse.go

```diff
@@ -0,0 +1,118 @@
+package variables
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/hcl"
+)
+
+// ParseInput parses a manually inputed variable to a richer value.
+//
+// This will turn raw input into rich types such as `[]` to a real list or
+// `{}` to a real map. This function should be used to parse any manual untyped
```

Comment:
@jbardin "untyped" in the sense it isn't an ast.Variable, its just an `interface{}` map. I'll make it clearer though.